File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/oauth2/docs/github/GithubUser.kt
package io.micronaut.security.oauth2.docs.github

//tag::clazz[]
import tools.jackson.databind.PropertyNamingStrategies
import tools.jackson.databind.annotation.JsonNaming
import io.micronaut.core.annotation.Introspected

@Introspected
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
class GithubUser {

    lateinit var login: String
    var name: String? = null
    var email: String? = null
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/security/oauth2/docs/github/GithubUser.py
# tag::clazz[]
from dataclasses import dataclass

from micronaut.core.annotation import Introspected
from tools.jackson.databind import PropertyNamingStrategies
from tools.jackson.databind.annotation import JsonNaming


@Introspected
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy)
@dataclass
class GithubUser:
    login: str | None = None
    name: str | None = None
    email: str | None = None
# end::clazz[]
